pkg/cloud/alibaba/assets/cloud-controller-manager-deployment.yaml

```diff
@@ -0,0 +1,100 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  labels:
+    app: alibaba-cloud-controller-manager
+  name: alibaba-cloud-controller-manager
+  namespace: openshift-cloud-controller-manager
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: alibaba-cloud-controller-manager
+  template:
+    metadata:
+      labels:
+        app: alibaba-cloud-controller-manager
+    spec:
+      hostNetwork: true
+      serviceAccountName: cloud-controller-manager
+      priorityClassName: system-cluster-critical
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: "kubernetes.io/hostname"
+              labelSelector:
+                matchLabels:
+                  app: alibaba-cloud-controller-manager
+      tolerations:
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
+          operator: Exists
+        - effect: NoExecute
+          key: node.kubernetes.io/unreachable
+          operator: Exists
+          tolerationSeconds: 120
+        - effect: NoExecute
+          key: node.kubernetes.io/not-ready
+          operator: Exists
+          tolerationSeconds: 120
+        - effect: NoSchedule
+          key: node.cloudprovider.kubernetes.io/uninitialized
+          operator: Exists
+        - effect: NoSchedule
+          key: node.kubernetes.io/not-ready
+          operator: Exists
+      containers:
+        - command:
```

Comment:
Please check the test results on this file as there are a number of requirements we set out via tests for the CCMs. The messages should hopefully be self explanatory, if they aren't, you can look to AWS as an example for what some of these might mean